controllers/solrcloud_controller.go

```diff
@@ -167,6 +167,21 @@ func (r *SolrCloudReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 				}
 			}
 		}
+	} else {
```

Comment:
This looks pretty straightforward to me. I'd say we should do it for the headless service as well, but honestly we might want to just always make a headless service for simplicity...